AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -1165,64 +1119,79 @@ protected override void OnMouseMove(MouseEventArgs e)
 
-                            StatisticsUpdated?.Invoke(this, EventArgs.Empty);
-                            break;
-                        case MouseMode.DragSelection:
-                            // move all selected objects
-                            var dx = (int)_coordinateHelper.ScreenToGrid(_mousePosition.X - _mouseDragStart.X, GridSize);
-                            var dy = (int)_coordinateHelper.ScreenToGrid(_mousePosition.Y - _mouseDragStart.Y, GridSize);
-                            // check if the mouse has moved at least one grid cell in any direction
-                            if (dx == 0 && dy == 0)
-                            {
-                                StatisticsUpdated?.Invoke(this, EventArgs.Empty);
+                                // adjust rect
+                                _selectionRect = new Rect(_mouseDragStart, _mousePosition);
+                                // select intersecting objects
+                                _selectedObjects.AddRange(_placedObjects.FindAll(_ => _.CalculateScreenRect(GridSize).IntersectsWith(_selectionRect)));
+
+                                StatisticsUpdated?.Invoke(this, UpdateStatisticsEventArgs.All);
                                 break;
                             }
-
-                            var unselected = _placedObjects.FindAll(_ => !_selectedObjects.Contains(_));
-                            var collisionsExist = false;
-                            // temporarily move each object and check if collisions with unselected objects exist
-                            foreach (var obj in _selectedObjects)
+                        case MouseMode.DragSelection:
                             {
-                                var originalPosition = obj.Position;
-                                // move object                                
-                                obj.Position = new Point(obj.Position.X + dx, obj.Position.Y + dy);
-                                // check for collisions
-                                var collides = unselected.Find(_ => ObjectIntersectionExists(obj, _)) != null;
-                                obj.Position = originalPosition;
-                                if (collides)
+                                // move all selected objects
+                                var dx = (int)_coordinateHelper.ScreenToGrid(_mousePosition.X - _mouseDragStart.X, GridSize);
+                                var dy = (int)_coordinateHelper.ScreenToGrid(_mousePosition.Y - _mouseDragStart.Y, GridSize);
+                                // check if the mouse has moved at least one grid cell in any direction
+                                if (dx == 0 && dy == 0)
                                 {
-                                    collisionsExist = true;
+                                    //no relevant mouse move -> no need for update
+                                    //StatisticsUpdated?.Invoke(this, EventArgs.Empty);
```

Comment:
What is the purpose of this commented out code - if we don't need it then remove it, or add some additional explanation as to why its there.

Reply:
I changed the comment to make it more clear.